cache/remotecache/s3/s3_downloader.go

```diff
@@ -0,0 +1,219 @@
+package s3
+
+// To be replaced by direct SDK call when available
+// https://github.com/aws/aws-sdk-go-v2/issues/2247
+
+import (
+	"context"
+	"io"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/containerd/containerd/content"
+	"golang.org/x/sync/errgroup"
+)
+
+type S3Downloader struct {
+	input *S3DownloaderInput
+}
+
+type S3DownloaderInput struct {
```

Comment:
These don't need to be public if only used by private functions.

Reply:
Done